File: RudderSDKCore/RudderSDKCore/RudderLogger.m
//
//  RudderLogger.m
//  RudderSDKCore
//
//  Created by Arnab Pal on 17/10/19.
//  Copyright © 2019 Rudderlabs. All rights reserved.
//

#import "RudderLogger.h"

static NSString *TAG = @"RudderSDKCore";
static int logLevel;

@implementation RudderLogger

- (instancetype)init
{
    self = [super init];
    if (self) {
        logLevel = 3;
    }
    return self;
}

+ (void)initiate:(int)_logLevel {
    if (_logLevel > 5) {
        logLevel = 5;
    } else if (_logLevel < 0) {
        logLevel = 0;
    } else {
        logLevel = _logLevel;
    }
}

+ (void)logVerbose:(NSString *)message {
    if (logLevel >= 5) {
        NSLog(@"%@:Verbose:%@", TAG, message);
    }
}

+ (void)logDebug:(NSString *)message {
    if (logLevel >= 4) {
        NSLog(@"%@:Debug:%@", TAG, message);
    }
}

+ (void)logInfo:(NSString *)message {
    if (logLevel >= 3) {
        NSLog(@"%@:Info:%@", TAG, message);
    }
}

+ (void)logWarn:(NSString *)message {
    if (logLevel >= 2) {
        NSLog(@"%@:Warn:%@", TAG, message);
    }
}

+ (void)logError:(NSString *)message {
    if (logLevel >= 1) {
        NSLog(@"%@:Error:%@", TAG, message);
    }
}


@end

File: RudderSDKCore/Classes/RudderLogger.swift
//
//  RudderLogger.swift
//  RudderSdkCore
//
//  Created by Arnab Pal on 09/10/19.
//  Copyright © 2019 Arnab Pal. All rights reserved.
//

import Foundation

class RudderLogger {
    private static var logLevel = RudderLogLevel.INFO;
    private static let TAG = "RudderSDK";
    
    init(_logLevel: Int) {
        if (_logLevel > RudderLogLevel.DEBUG) {
            RudderLogger.logLevel = RudderLogLevel.DEBUG;
        } else if (_logLevel < RudderLogLevel.NONE) {
            RudderLogger.logLevel = RudderLogLevel.NONE;
        } else {
            RudderLogger.logLevel = _logLevel;
        }
    }
    
    static func logError(message: String) {
        if (RudderLogger.logLevel >= RudderLogLevel.ERROR) {
            print(TAG + ":Error:" + message);
        }
    }
    
    static func logError(error: Error) {
        if (RudderLogger.logLevel >= RudderLogLevel.ERROR) {
            print(TAG + ":Error:")
            print(error);
        }
    }
    
    static func logWarn(message: String) {
        if (RudderLogger.logLevel >= RudderLogLevel.WARN) {
            print(TAG + ":Warn:" + message);
        }
    }
    
    static func logInfo(message: String) {
        if (RudderLogger.logLevel >= RudderLogLevel.INFO) {
            print(TAG + ":Info:" + message);
        }
    }
    
    static func logDebug(message: String) {
        if (RudderLogger.logLevel >= RudderLogLevel.DEBUG) {
            print(TAG + ":Debug:" + message);
        }
    }
}

class RudderLogLevel {
    static let DEBUG: Int = 4;
    static let INFO: Int = 3;
    static let WARN: Int = 2;
    static let ERROR: Int = 1;
    static let NONE: Int = 0;
}
